Add k-NN radial search doc (#7050)
opensearch-project/documentation-website: _query-dsl/specialized/neural.md

`query_text` | String | Optional | The query text from which to generate vector embeddings. You must specify at least one `query_text` or `query_image`.
`query_image` | String | Optional | A base-64 encoded string that corresponds to the query image from which to generate vector embeddings. You must specify at least one `query_text` or `query_image`.
`model_id` | String | Required if the default model ID is not set. For more information, see [Setting a default model on an index or field]({{site.url}}{{site.baseurl}}/search-plugins/neural-text-search/#setting-a-default-model-on-an-index-or-field). | The ID of the model that will be used to generate vector embeddings from the query text. The model must be deployed in OpenSearch before it can be used in neural search. For more information, see [Using custom models within OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/) and [Neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/).

> `k` | Integer | Optional | The number of results returned by the k-NN search. Default is 10. Only one of `k` or `min_score` or `max_distance` can be specified.

Reviewer: X = 2.14

> `k` | Integer | Optional | The number of results returned by the k-NN search. Default is 10. Only one of `k` or `min_score` or `max_distance` can be specified.

Author: @navneet1v In previous versions, the `k` wasn't mandatory either. Default 10 is applied if no `k` is provided. With compatibility, same pattern applied from 2.14 version, if no of `k`, `min_score` or `max_distance` is provided, `k` with default 10 is applied in the neural search.

> `k` | Integer | Optional | The number of results returned by the k-NN search. Default is 10. Only one of `k` or `min_score` or `max_distance` can be specified.

Reviewer 2: thanks @junqiu-lei for providing the info. I might have forget about this. But for k-NN it `k` is now becoming optional. So can we add that line which says k is now optional starting from version 2.14

> `k` | Integer | Optional | The number of results returned by the k-NN search. Default is 10. Only one of `k` or `min_score` or `max_distance` can be specified.

Author: @navneet1v Updated in approximate-knn.md

> `k` | Integer | Optional | The number of results returned by the k-NN search. Default is 10. Only one of `k` or `min_score` or `max_distance` can be specified.

Author: Thanks @Naarcha-AWS, `min_score` and `max_distance` are not belongs to `k`. User can specify one of `k`, `min_score`, `max_distance` to use.

> `min_score` | Float | Optional | The minimum score threshold for the search results. Only one of `k` or `min_score` or `max_distance` can be specified. See [k-NN radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/) for more information.

Author: Same above

> `max_distance` | Float | Optional | The maximum distance threshold for the search results. Only one of `k` or `min_score` or `max_distance` can be specified. See [k-NN radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/) for more information.

Author: Same above

`query_text` | String | Optional | The query text from which to generate vector embeddings. You must specify at least one `query_text` or `query_image`.
`query_image` | String | Optional | A base-64 encoded string that corresponds to the query image from which to generate vector embeddings. You must specify at least one `query_text` or `query_image`.
`model_id` | String | Required if the default model ID is not set. For more information, see [Setting a default model on an index or field]({{site.url}}{{site.baseurl}}/search-plugins/neural-text-search/#setting-a-default-model-on-an-index-or-field). | The ID of the model that will be used to generate vector embeddings from the query text. The model must be deployed in OpenSearch before it can be used in neural search. For more information, see [Using custom models within OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/) and [Neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/).

> `k` | Integer | Optional | The number of results returned by the k-NN search. Only one variable, either `k`, `min_score`, or `max_distance`, can be specified. If a variable is not specified, the default is `k` with a value `10`.

Reviewer 3: ```suggestion
`k` | Integer | Optional | The number of results returned by the k-NN search. Only one variable, either `k`, `min_score`, or `max_distance`, can be specified. If a variable is not specified, the default is `k` with a value of `10`.
```

> `min_score` | Float | Optional | The minimum score threshold for the search results. Only one variable, either `k`, `min_score` or `max_distance`, can be specified. For more information, see [k-NN radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/)

Reviewer 3: ```suggestion
`min_score` | Float | Optional | The minimum score threshold for the search results. Only one variable, either `k`, `min_score`, or `max_distance`, can be specified. For more information, see [k-NN radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/).
```

`max_distance` | Float | Optional | The maximum distance threshold for the search results. Only one variable, either `k`, `min_score`, or `max_distance`, can be specified. For more information, see [k-NN radial search]({{site.url}}{{site.baseurl}}/search-plugins/knn/radial-search-knn/).
`filter` | Object | Optional | A query that can be used to reduce the number of documents considered. For more information about filter usage, see [k-NN search with filters]({{site.url}}{{site.baseurl}}/search-plugins/knn/filter-search-knn/). **Important**: Filter can only be used with the `faiss` or `lucene` engines.

#### Example request

> The following example shows a search with a top 100 k-NN and a filter that includes a range query and a term query:

Reviewer 3: Can we clarify what is meant by "a top 100 k-NN"? What is that?

Reviewer 4: It means it'll look for the first or top 100 rows or columns in a search, depending on the search. It should be understood by more consuming the content. 

How about "a k-NN value of `100`"?

Reviewer 3: That works. Thanks!

> The following example shows a search with a top 100 k-NN and a filter that includes a range query and a term query:

Reviewer 4: ```suggestion
The following example shows a search with a `k` value of `100` and a filter that includes a range query and a term query:
```
